template/9-design.md

```diff
@@ -1,56 +1,193 @@
+- **Description**: The QR Code screen facilitates quick and easy friend additions. The screen displays the user's unique QR code, which can be scanned by others to add them as a friend and activates the camera to scan other users' QR codes. This screen ensures that users can effortlessly expand their network by leveraging QR code technology.
+#### 6 Event Details Screen
+- **Features**: Displays detailed information about a selected event, including date, time, location, and group chat (not in POC)
+- **Components**: Event description, group chat redirection button
+- **Description**: The Event Details Screen offers comprehensive details about the selected event, enhancing the user's understanding of the event's key aspects. Additionally, registered users can access the event's group chat (not available in POC), facilitating communication and engagement among attendees.
+### Backend Communication
+The app will utilize Firebase for backend communication, leveraging its robust set of tools and services:
+- **Authentication**: Firebase Authentication for secure login and user management.
+- **Database**: Firebase Firestore for real-time data synchronization and storage.
+- **Cloud Storage**: Firebase Storage for securely storing user-uploaded photos and other media.
 
-*List the key libraries, languages, components used by the MVP.*
-- Key Libraries/Languages: Kotlin, Hilt, Material3, Jetpack Compose, Firebase, Android Camera Image Analysis.
-
-*If applicable, describe essential screens.*
-- Home screen, Event Details, User Profile, Messaging, Event Creation.
 ## Backend
-
-*Decompose the MVP into functional blocks.*
-- User Management: Handles registrations, profiles, and settings.
-- Event Management: Creates and updates events, manages tickets.
-- Messaging System: Manages communication between users.
+### User Management
+- **Description**: This functional block handles user-related operations such as registrations, profiles, and interactions between users (e.g., becoming friends).
+- **Application Logic**: Manages user authentication, registration, profile updates, and friend requests.
+- **Framework**: Developed using Firebase Authentication and Firestore for data storage.
+- **Dynamic View Rendering**: Integrates with the frontend to render user profiles and friend lists dynamically based on data from Firestore.
+- **Responsive Backend-Frontend Interaction**: Ensures responsive interaction with the frontend, updating user profiles and friend lists based on user interactions.
+### Event Management
+- **Description**: This block handles operations related to creating, updating, and managing events, including ticket management.
+- **Application Logic**: Manages event creation, updates, and ticket sales.
+- **Framework**: Utilizes Firebase Firestore for event data storage.
+- **Dynamic View Rendering**: Provides APIs to fetch event details dynamically, ensuring frontend displays reflect the latest event information.
```

Comment:
add something about payment API?

Reply:
Modified as well to take that into account